src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -119,120 +134,170 @@ static Schema preProcessSchema(Schema schema) {
         case Time.LOGICAL_NAME:
         case Timestamp.LOGICAL_NAME:
           return schema;
+        default:
+          // User type or unknown logical type
+          break;
       }
     }
 
     Schema.Type schemaType = schema.type();
     switch (schemaType) {
-      case ARRAY: {
-        return copySchemaBasics(schema, SchemaBuilder.array(preProcessSchema(schema.valueSchema()))).build();
-      }
-      case MAP: {
-        Schema keySchema = schema.keySchema();
-        Schema valueSchema = schema.valueSchema();
-        String keyName = keySchema.name() == null ? keySchema.type().name() : keySchema.name();
-        String valueName = valueSchema.name() == null ? valueSchema.type().name() : valueSchema.name();
-        if (keySchema.type() == Schema.Type.STRING) {
-          return SchemaBuilder.map(preProcessSchema(keySchema), preProcessSchema(valueSchema)).build();
-        }
-        Schema elementSchema = SchemaBuilder.struct().name(keyName + "-" + valueName)
-             .field(MAP_KEY, preProcessSchema(keySchema))
-             .field(MAP_VALUE, preProcessSchema(valueSchema))
-             .build();
-        return copySchemaBasics(schema, SchemaBuilder.array(elementSchema)).build();
-      }
-      case STRUCT: {
-        SchemaBuilder structBuilder = copySchemaBasics(schema, SchemaBuilder.struct().name(schemaName));
-        for (Field field : schema.fields()) {
-          structBuilder.field(field.name(), preProcessSchema(field.schema()));
-        }
-        return structBuilder.build();
-      }
-      default: {
+      case ARRAY:
+        return preProcessArraySchema(schema);
+      case MAP:
+        return preProcessMapSchema(schema);
+      case STRUCT:
+        return preProcessStructSchema(schema);
+      default:
         return schema;
-      }
     }
   }
 
+  private static Schema preProcessArraySchema(Schema schema) {
+    Schema valSchema = preProcessSchema(schema.valueSchema());
+    return copySchemaBasics(schema, SchemaBuilder.array(valSchema)).build();
+  }
+
+  private static Schema preProcessMapSchema(Schema schema) {
+    Schema keySchema = schema.keySchema();
+    Schema valueSchema = schema.valueSchema();
+    String keyName = keySchema.name() == null ? keySchema.type().name() : keySchema.name();
+    String valueName = valueSchema.name() == null
+                       ? valueSchema.type().name()
```

Comment:
nit: My impression was that these split on literals would be like indenting consecutive method calls. Aligned at the very start of the line, 2 tabs in (== 4 spaces in this repo's settings). Never mind if that's not the case.